src/main/java/commission/dao/SalesCommissionDao.java

```diff
@@ -0,0 +1,15 @@
+package commission.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface SalesCommissionDao<T>{
+
+    List<T> findAll();
+    Optional<T> findById(long id);
+    void insert(T entity);
+    void update(long id, T entity);
+    void delete(long id);
+
+
+}
```

Comment:
Cool that you are using generics! This is general CRUD functionality and not really related to sales or commission. I would rather name this more generic like `CrudDao` or `AbstractDao` or `BaseDao`

Reply:
Got it, those names are much better.  
PS: thank you, this is a result of many things that I've learned from you. Isn't this Abstraction from OOP?

Reply:
Yup this is abstraction, however abstraction is about grouping together common functionality or related classes. That's why I suggested the change here, because this isn't "Sale" or "Commission" related but rather CRUD operations that are being abstracted. This allows you to have a definition of a "CRUD" DAO in your interface that can be implemented by any DAO that requires this functionality.

Some other types of examples:
You could have an abstract class for "Sale" and have two child classes that could be "CommissionSale" and "FeeSale". These could have similar properties / variables like `saleAmount`, `salesPerson`, `saleDate` etc. But have an abstract method `calculateCommission` that is implemented differently in the two classes (which now incorporates polymorphism into the mix).

Alternative you can also use an interface for different types of commission:
```
interface CommissionEnabled {
  BigDecimal calculateCommission();
}
```

You can then implement this interface in both different commission classes. What that allows you to do is write code like this:
```
var salesPerson = dao.find(...);
List<CommissionEnabled> sales = findSales(salesPerson);
for (CommissionEnabled sale: sales) {
  salesPerson.addCommission(sale.calculateCommission()); // Implementation of wether it's a commissioned sale, fee sale etc does not matter. We just know that every implementation will have a "calculateCommission" method, but how it calculates the commission is irrelevant that is up to the class implementing the interface.
}
```